src/app/scripts/cac/routing/cac-routing-itinerary.js

```diff
@@ -195,6 +197,48 @@ CAC.Routing.Itinerary = (function ($, cartodb, L, _, moment, Geocoder, Utils) {
             leg.formattedDuration = getFormattedDuration(leg);
             return leg;
         });
+        if (hasWaypoints) {
+            newLegs = mergeLegsAcrossWaypoints(newLegs);
+        }
+        return newLegs;
+    }
+
+    /* Waypoints always result in a step break, which ends up producing useless "To Destination"
+     * steps in the itinerary details.
+     * See https://github.com/opentripplanner/OpenTripPlanner/blob/otp-1.0.0/src/main/java/org/opentripplanner/routing/impl/GraphPathFinder.java#L305
+     * and https://github.com/opentripplanner/OpenTripPlanner/blob/otp-1.0.0/src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java#L212
+     * There's no configuration option to make OTP not do that, so instead this munges the
+     * resulting separate steps into one.
+     * Specifically, it loops over the legs, checking for each one whether the next one is the same
+     * mode and, if so, summing times/distances and resetting the 'to' to turn the first leg into
+     * a combination of itself and the second.  Since there can be multiple waypoints in what would
+     * be a single leg, more than two consecutive legs can end up getting merged together.
+     *
+     * Note that this makes no attempt to merge the `legGeometry` attributes so it's important that
+     * `getFeatures`, which gets the itinerary's geometry into Leaflet, gets called on the original
+     * legs array rather than the munged one.
+     */
+    function mergeLegsAcrossWaypoints(legs) {
+        if (legs.length === 1) {
+            return legs;
+        }
+        var index = 0;
+        while(index < legs.length - 1) {
```

Comment:
Running a `while` loop over a mutating collection makes me a little nervous, but I can't think of any cases where this might get stuck in an infinite loop or throw an out-of-range exception.